### PR TITLE
Add xiao-ble-battery binding and device tree objects

### PR DESCRIPTION
**Pull request by DBS06**
to to use your great library more in a zephyr specific way.

With these changes it is possible to configure your library without touching the library itself.

I hope you will like it

You have done a great work btw!

### CODE REVIEW

File: boards/xiao_ble_nrf52840.overlay [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/ {
+	xio_ble_battery_dev: xio_ble_battery_dev {
```
Comment (Tjoms99) [inline boards/xiao_ble_nrf52840.overlay:2]: This is probably a typo, should be xiao_ble_battery_dev?

File: boards/xiao_ble_nrf52840_sense.overlay [hunk 2]
```diff
@@ -0,0 +1,9 @@
+/ {
+	xio_ble_battery_dev: xio_ble_battery_dev {
```
Comment (Tjoms99) [inline boards/xiao_ble_nrf52840_sense.overlay:2]: Same here

File: src/battery/battery.c [hunk 3]
```diff
@@ -22,33 +22,35 @@
 #include <zephyr/drivers/gpio.h>
 #include <zephyr/drivers/adc.h>
 #include <zephyr/logging/log.h>
+
 LOG_MODULE_REGISTER(battery, LOG_LEVEL_INF);
 
-#define GPIO_BATTERY_CHARGE_SPEED 13
-#define GPIO_BATTERY_CHARGING_ENABLE 17
-#define GPIO_BATTERY_READ_ENABLE 14
+#if !DT_NODE_EXISTS(DT_NODELABEL(xio_ble_battery_dev))
```
Comment (Tjoms99) [inline src/battery/battery.c:28]: Same here

File: src/battery/battery.c [hunk 4]
```diff
@@ -22,33 +22,35 @@
...
+#error "Overlay for xio_ble_battery_dev node not properly defined."
+#endif
 
-// Feel free to increase this when necessary
-#define BATTERY_CALLBACK_MAX 3
+#define BATTERY_NODE DT_NODELABEL(xio_ble_battery_dev)
```
Comment (Tjoms99) [inline src/battery/battery.c:32]: Same here

**Review by Tjoms99 — COMMENTED** [review]
This is awesome, thank you for the contribution! Just commented on a minor thing, but other than that it looks good

**Comment by DBS06** [comment]
Yeah, it's definitely a typo 😅
I will fix it on Monday when I am back at work 😉
I am glad that you liked my changes!

**Review by Tjoms99 — APPROVED** [review]

**Pull request merged**